client/sites-dashboard/components/sites-dashboard-opt-in-banner.tsx

```diff
@@ -0,0 +1,110 @@
+import TrackComponentView from 'calypso/lib/analytics/track-component-view';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { useAsyncPreference } from 'calypso/state/preferences/use-async-preference';
+import {
+	SITES_AS_LANDING_PAGE_DEFAULT_VALUE,
+	SITES_AS_LANDING_PAGE_PREFERENCE,
+} from 'calypso/state/sites/selectors/has-sites-as-landing-page';
+
+interface SitesDashboardOptInBannerProps {
+	sites: SiteExcerptData[];
+}
+
+const SitesNotice = styled( Notice )( {
+	marginBlockStart: '32px',
+	backgroundColor: 'var(--color-neutral-0) !important',
+	color: 'var(--color-text) !important',
+
+	'.notice__icon-wrapper': {
+		paddingInlineStart: '8px',
+		background: 'transparent !important',
+		color: 'var(--color-accent) !important',
+	},
+
+	'.notice__content': {
+		paddingBlock: '16px',
+		paddingInline: '8px',
+		display: 'flex',
+		alignItems: 'center',
+	},
+} );
+
+const NoticeActions = styled.div( {
+	display: 'flex',
+	gap: '16px',
+	alignItems: 'center',
+	padding: '16px',
+	flexShrink: 0,
+} );
+
+const hasDismissedTheBannerRecently = ( {
+	updatedAt,
+}: typeof SITES_AS_LANDING_PAGE_DEFAULT_VALUE ) => {
+	const fourteenDaysAgo = moment().subtract( 14, 'days' ).startOf( 'day' );
+
+	return moment( updatedAt ).isAfter( fourteenDaysAgo );
+};
+
+export const SitesDashboardOptInBanner = ( { sites }: SitesDashboardOptInBannerProps ) => {
+	const isSmallScreen = useBreakpoint( '<660px' );
+	const { __ } = useI18n();
+	const dispatch = useDispatch();
+	const [ landingPagePreference, setLandingPagePreference ] = useAsyncPreference( {
+		defaultValue: SITES_AS_LANDING_PAGE_DEFAULT_VALUE,
+		preferenceName: SITES_AS_LANDING_PAGE_PREFERENCE,
+	} );
+
+	if (
+		sites.length < 2 ||
+		landingPagePreference === 'none' ||
+		landingPagePreference.useSitesAsLandingPage ||
+		hasDismissedTheBannerRecently( landingPagePreference )
+	) {
+		return null;
+	}
+
+	const handleAccept = () => {
+		setLandingPagePreference( {
+			useSitesAsLandingPage: true,
+			updatedAt: new Date().valueOf(),
+		} );
+		dispatch( recordTracksEvent( 'calypso_sites_as_landing_page_accepted' ) );
+	};
+
+	const handleDismiss = () => {
+		setLandingPagePreference( {
+			useSitesAsLandingPage: false,
+			updatedAt: new Date().valueOf(),
+		} );
+		dispatch( recordTracksEvent( 'calypso_sites_as_landing_page_rejected' ) );
+	};
+
+	return (
+		<>
+			<TrackComponentView eventName="calypso_sites_as_landing_page_seen" />
+			<SitesNotice
+				status="is-info"
+				text={ __( 'Do you want to make this page your default when visiting WordPress.com?' ) }
+				icon="heart"
+				showDismiss={ false }
+			>
+				<NoticeActions>
+					<Button borderless onClick={ handleDismiss }>
+						{ isSmallScreen ? __( 'No' ) : __( 'No, thanks' ) }
+					</Button>
+					<Button onClick={ handleAccept }>
+						{ isSmallScreen ? __( 'Yes' ) : __( 'Yes, make it my home' ) }
+					</Button>
+				</NoticeActions>
```

Comment:
I think putting the primary button on the left makes a bit more sense for LTR language:

<img width="1325" alt="image" src="https://user-images.githubusercontent.com/36432/204387829-2b514ff2-4dca-402f-acbf-019fa28ff278.png">

Also, should there be a confirmation message once you hit "Yes"? I was surprised to have the notice disappear and then nothing happen...

Reply:
This makes sense, probably a notification pop-up stating sites is now default landing page might suffice.

Reply:
> I think putting the primary button on the left makes a bit more sense for LTR language

I've raised that here: p1669686068039569-slack-C041RHH38NQ

> should there be a confirmation message once you hit "Yes"?

I'm wondering whether it wasn't just the lack of confirmation but also whether the big layout shift that happens when the banner disappears was also part of what felt surprising.

Reply:
> I think putting the primary button on the left makes a bit more sense for LTR language

We should work towards a standard pattern where primary action is on right.

![image](https://user-images.githubusercontent.com/1072756/204447833-361e888a-dab6-4103-bea7-42e514c28e96.png)

> Also, should there be a confirmation message once you hit "Yes"?

Good use case for [Snackbar](https://wordpress.github.io/gutenberg/?path=/story/components-snackbar--default) or [GlobalNotice](https://wpcalypso.wordpress.com/devdocs/design/global-notices).